### PR TITLE
Change the backing data structure of VarSet from slice to map

### PR DESCRIPTION
**Pull request by ian-howell**
This will speed up most operations performed on a large set of Vars

More detailed information in [this issue](https://github.com/kubernetes-sigs/kustomize/issues/1223).

### CODE REVIEW

**Comment by ian-howell** [comment]
I believe travis has croaked on this one. Closing and reopening to trigger a new build

File: pkg/types/var.go [hunk 1]
```diff
@@ -70,26 +70,45 @@ func (v *Var) defaulting() {
...
 // AsSlice returns the vars as a slice.
 func (vs *VarSet) AsSlice() []Var {
 	s := make([]Var, len(vs.set))
-	copy(s, vs.set)
+	i := 0
+	for _, v := range vs.set {
+		s[i] = v
+		i++
+	}
+	sort.Sort(ByName(s))
```
Comment (monopole) [inline pkg/types/var.go]: Please toss in a String() method that calls this method so we get deterministic ordering when debugging.

**Comment by monopole** [comment]
nevermind, if it becomes an issue, will add the String() 
thanks!

**Pull request merged**